### ISSUE TITLE
Panic on CentOS 6.10 reading kprobe events

### ISSUE THREAD

**Issue by andrewkroh** [andrewkroh/perf#1]
```
$ uname -a
Linux centos-6 2.6.32-754.9.1.el6.x86_64 #1 SMP Thu Dec 6 08:02:15 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
$ cat /etc/redhat-release 
CentOS release 6.10 (Final)
```

```
panic: runtime error: slice bounds out of range

goroutine 7 [running]:
acln.ro/perf.(*Event).readRawRecordNonblock(0xc00007d080, 0xc000068f58, 0x2)
	/Users/akroh/go/pkg/mod/github.com/andrewkroh/perf@v0.0.0-20190607173508-cf8284842038/record.go:200 +0x2a1
acln.ro/perf.(*Event).ReadRawRecord(0xc00007d080, 0x53b920, 0xc000044200, 0xc000068f58, 0x0, 0xc00007d080)
	/Users/akroh/go/pkg/mod/github.com/andrewkroh/perf@v0.0.0-20190607173508-cf8284842038/record.go:148 +0x22a
main.main.func1(0xc00007cf20, 0x53b920, 0xc000044200)
	/Users/akroh/go/src/github.com/andrewkroh/go-ebpf/perf/main.go:98 +0x1fd
created by main.main
	/Users/akroh/go/src/github.com/andrewkroh/go-ebpf/perf/main.go:74 +0x536
```

**Comment by adriansr** [comment]
This is caused by the perf event channel returning only zeroed memory and `readRawRecordNonblock` not being prepared for a header size of zero.

**Comment by adriansr** [comment]
I have seen this problem in 3.16 (Debian 8) too.

Seems to be a disagreement in the record format in use. Header.Size is 0 and `readRawRecordNonblock` doesn't correctly check this before slicing the header away (8 bytes).

It has to be a regression somewhere. Older 3.x kernels do it right. I'm trying to figure out if some arguments to `perf_event_open` need tweaking.

**Comment by adriansr** [comment]
Seems that there's a problem with the ring buffer that is used to communicate with the kernel. The values on unix.PerfEventMmapPage are not populated so the usermode is reading the ringbuffer from a bogus location

**Comment by adriansr** [comment]
Here's the fix

https://github.com/adriansr/socket_tracer/commit/751dcf5ae0d5afac8c3c1805e8994550d302333c

**Comment by andrewkroh** [comment]
Fixed 2c5ccac8c82bb75b52e5461d5de8248572165106 and PR was submitted upstream.
